webapp/commonComponents/expression/nodes/binary.js

```diff
@@ -1,37 +1,59 @@
 import React from 'react'
+import PropTypes from 'prop-types'
 import * as R from 'ramda'
 
 import * as Expression from '@core/expressionParser/expression'
+import * as NodeDef from '@core/survey/nodeDef'
 
 import Dropdown from '../../form/dropdown'
 import BinaryOperand, { BinaryOperandType } from './binaryOperand'
 import EditButtons from './editButtons'
 
-const Binary = props => {
-  const { node, onChange, canDelete = false, onDelete, isBoolean } = props
+const Binary = (props) => {
+  const { canDelete, node, nodeDefCurrent, isBoolean, onChange, onDelete } = props
 
   const isLeftLiteral = R.pipe(R.prop(BinaryOperandType.left), Expression.isLiteral)(node)
 
+  const showOperator = (isBoolean && !NodeDef.isBoolean(nodeDefCurrent)) || !isLeftLiteral
+
   return (
     <div className="binary">
-      <BinaryOperand {...props} type={BinaryOperandType.left} />
+      {React.createElement(BinaryOperand, { ...props, type: BinaryOperandType.left })}
```

Comment:
use <BinaryOperand /> with explicit props

Reply:
it will be a bit more unreadable but ok, I'll do it

Reply:
I meant to use <BinaryOperand prop1={} ...